tests/TestCase/View/Helper/FormHelperTest.php

```diff
@@ -8483,4 +8483,77 @@ public function testFormValuesSourcesResetViaEnd()
         $result = $this->Form->getValuesSources();
         $this->assertEquals($expected, $result);
     }
+
+    /**
+     * Test sources values schema defaults handling
+     *
+     * @return void
+     */
+    public function testFormValuesSourcesSchemaDefaults()
+    {
+        $Articles = TableRegistry::get('Articles');
+        $entity = $Articles->newEntity();
+        $this->Form->create($entity);
+        $result = $this->Form->getSourceValue('title');
+        $expected = '';
+        $this->assertEquals($expected, $result);
+
+        $Articles = TableRegistry::get('Articles');
+        $title = $Articles->schema()->column('title');
+        $Articles->schema()->addColumn(
+            'title',
+            ['default' => 'default title'] + $title
+        );
+        $entity = $Articles->newEntity();
+        $this->Form->create($entity);
+        $result = $this->Form->getSourceValue('title');
+        $expected = 'default title';
+        $this->assertEquals($expected, $result);
+
+        $Articles = TableRegistry::get('Articles');
+        $title = $Articles->schema()->column('title');
+        $Articles->schema()->addColumn(
+            'title',
+            ['default' => 'default title'] + $title
+        );
+        $entity = $Articles->newEntity();
+        $this->Form->create($entity);
+        $this->Form->setValuesSources(['query']);
+        $result = $this->Form->getSourceValue('title');
+        $expected = '';
+        $this->assertEquals($expected, $result);
```

Comment:
Because you can supply 'schemaDefaults' as an option to `FormHelper::create()` but not utilise the EntityContext? 

You say I should remove
```
        if (isset($options['schemaDefault'])) {
              return $options['schemaDefault'];
          }
```
?